pennylane_qiskit/qiskit_device.py

```diff
@@ -196,7 +221,14 @@ def apply(self, operations, **kwargs):
                 measure(self._circuit, qr, cr)
 
         # These operations need to run for all devices
-        qobj = self.compile()
+        temp_transpile_args = {}
+        transpile_sig = inspect.signature(transpile).parameters
+        self.set_args(
+            kwargs,
+            transpile_sig,
+            (self.transpile_args if persist_transpile_options else temp_transpile_args),
+        )
+        qobj = self.compile(temp_transpile_args)
```

Comment:
One thing to note for `apply` is that it is being called in the [`QubitDevice.execute` method](https://github.com/PennyLaneAI/pennylane/blob/a87ecbfff517fc7c8f736ed97d69625471497095/pennylane/_qubit_device.py#L169) by passing pre-defined arguments. This means that one would not be able to explicitly pass the `persist_transpile_options` option to `apply`.

There could be two ways to handle this behaviour:
1. We could simply resort to the transpile arguments being defined on device initialization
2. We could have a setter method for the transpile arguments (one that can update `dev.transpile_args`)

The second option would be nice to have, but it is also already nice to be able to define transpile options on device initialization.

Reply:
Created a setter for transpilation options. This will not merge the transpile parameters, but will override them.